pkg/distributor/distributor.go

```diff
@@ -637,6 +637,13 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 
 		if mrc := d.limits.MetricRelabelConfigs(userID); len(mrc) > 0 {
 			l := relabel.Process(cortexpb.FromLabelAdaptersToLabels(ts.Labels), mrc...)
+			if l == nil {
+				// all labels are gone, therefore the __name__ label is not present, metric will be discarded
```

Comment:
A better test for "all labels are gone" would be `len(l) == 0`.  I haven't investigated whether the code or the comment should be changed.

Reply:
Yeah, the underlying code from `relabel.Process` returns nil (so I thought it was okay to compare with nil), but `len` will catch empty slices as well, so I think it is better.